File: FishTrackerLambda/src/FishTrackerLambda/Functional/Wrapper.cs
using System;
namespace FishTrackerLambda.Functional
{
    public struct Wrapper<T>
	{
		bool m_valid = false;

		T? m_value;

		public Wrapper()
		{
			m_valid = false;

			m_value = default(T);
		}

		public Wrapper(T initial)
		{
			m_value = initial;

			m_valid = m_value != null;
		}

        public Wrapper(T initial, bool valid)
        {
            m_value = initial;

            m_valid = valid;
        }

        public static explicit operator Wrapper<T>(T value)
        {
			return new Wrapper<T>(value);
        }

		public bool IsOk => m_valid;
    }
}

File: FishTrackerLambdaNode/src/Functional/Wrapper.ts
export class Wrapper<T> {
    private m_valid: boolean = false;
    private m_value: T | null;

    constructor();
    constructor(initial: T);
    constructor(initial: T, valid: boolean);
    constructor(initial?: T, valid?: boolean) {
        if (initial !== undefined && valid !== undefined) {
            this.m_value = initial;
            this.m_valid = valid;
        } else if (initial !== undefined) {
            this.m_value = initial;
            this.m_valid = initial !== null;
        } else {
            this.m_value = null;
            this.m_valid = false;
        }
    }

    static explicit<T>(value: T): Wrapper<T> {
        return new Wrapper<T>(value);
    }

    get isOk(): boolean {
        return this.m_valid;
    }
}